File: Data Structures & Algorithms/binary-tree-diameter/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root) {
        var max_ = 0;

        function dfs(root){
            if (!root){
                return 0;
            }

            let left = dfs(root.left);
            let right = dfs(root.right);
            max_ = Math.max(max_, left + right)

            return Math.max(left, right) + 1
        }
        dfs(root)
        return max_
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        left =  0
        right = len(numbers) - 1

        while left < right:
            sums = numbers[left] + numbers[right]

            if target == sums:
                return [left + 1, right + 1]
            elif target > sums:
                left += 1
            elif target < sums:
                right -= 1

        

        


